plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt

```diff
@@ -857,16 +820,17 @@ class CameraAnimationsPluginImpl : CameraAnimationsPlugin {
         interpolator = it
       }
       animationOptions?.animatorListener?.let {
-        // duration == 0L does not behave the same as > 0 and actually
-        // our custom `highLevelListener` designed for proper notifying of lifecycle
-        // based on `runningAnimatorsQueue` is not working because animators end immediately.
-        // We use standard listener in that corner case.
-        highLevelListener = if (animationOptions.duration == 0L) {
-          addListener(it)
-          null
-        } else {
-          it
-        }
+        addListener(object : AnimatorListenerAdapter() {
+          override fun onAnimationEnd(animation: Animator?) {
+            if (commitScheduled) handler.removeCallbacks(commitChangesRunnable)
+            commitChangesRunnable.run()
+
+            if (highLevelAnimatorSet?.animatorSet === animation) {
```

Comment:
Can't get why we need it, can you explain a bit?

Reply:
It seems possible that the `onAnimationEnd` call for the canceled high level animation will occur after the new high level animation is started. To make sure `highLevelAnimatorSet` is valid it should be cleared only when the exact same animation is complete.